app/config/windows.js

```diff
@@ -8,11 +9,28 @@ const cfg = new Config({
   }
 });
 
+function validateAndFixWindowPosition(position, size) {
+  const displays = electron.screen.getAllDisplays();
+  const [x, y] = position;
+  const positionIsValid = displays.some(({workArea}) => {
```

Comment:
We can't test with any of available displays, only with targeted screen.
Screen targeting and some verification are done here: https://github.com/zeit/hyper/blob/5d0c6355e12ec8d9a89182556fa73227374b488f/app/index.js#L113-L148

It is a little bit messy and we probably should refactor it to add your position/size validation.

Reply:
>We can't test with any of available displays, only with targeted screen.

I'm having trouble understanding what you mean by this?

>It is a little bit messy and we probably should refactor it to add your position/size validation.

Sounds reasonable. Would you have the logic persist a fixed location? Or just assume that a correct location will be written on next close, and if not, the verification logic will take care of it again.

Reply:
My first point: I misunderstood that a windows referential is always the same. I was thinking that we specified a screen when opening a new window. I was wrong.

My second point: I would assume that a correct location will be written or fixed if needed. But never mind, let's merge this as it is.